radicle-node/src/test/tests.rs

```diff
@@ -397,7 +397,7 @@ fn test_push_and_pull() {
         &repo,
         "alice",
         "alice's repo",
-        storage::BranchName::from("master"),
+        storage::BranchName::try_from("master").unwrap(),
```

Comment:
Can we avoid this by having `BranchName::from(refname!("master"))` for eg.? Or perhaps we should just use `RefString` instead of `BranchName`?

Reply:
It looks like BranchName was already type alias'ed to git::refstring in the PR.

I prefer using 'try_from(..).unwrap(..)' for the tests.  Given we can panic (unwrap) happily in tests, so I believe it is ok.  Type support for try_from(..) is common, and we can reduce complexity by consistently using try_from(string) instead of different (string -> some_type) methods for different types.

Reply:
Updated to use git::refname!(..)